scripts/format-rewrites.py

```diff
@@ -0,0 +1,136 @@
+import sys
+from collections.abc import Iterable, Iterator
+from io import StringIO
+
+
+def main() -> None:
+    for line in sys.stdin:
+        print(_process_line(line.rstrip()))
+
+
+def _process_line(line: str) -> str:
+    prefix = "  |"
+    if line.startswith(prefix):
+        tokens = _tokenize(line)
+        return CtorWriter(tokens).print()
+    return line
+
+
+def _tokenize(text: str) -> list[str]:
+    import re
+
+    pattern = r"{|}|\(|\)|\||:=|:|,|«[^»]+»|[^{}()|:,«»\s]+"
```

Comment:
```suggestion
    pattern = r"{|}|\(|\)|\||:=|:|,|[^\s{}()|:,«»]*\.«[^»]+»|[^\s{}()|:,«»]*«[^»]+»|[^\s{}()|:,«»]+"
```
I find this RE works well and doesn't cause the above mentioned spaces